checkstyle/import-control.xml

```diff
@@ -69,6 +69,17 @@
       <allow pkg="org.apache.kafka.common.metrics" />
     </subpackage>
 
+    <!-- Third-party compression libraries should only be references from this package -->
+    <subpackage name="compression">
+      <allow pkg="com.github.luben.zstd" />
+      <allow pkg="net.jpountz.lz4" />
+      <allow pkg="net.jpountz.xxhash" />
+      <allow pkg="org.apache.kafka.common.compression" />
+      <allow pkg="org.xerial.snappy" />
+      <!-- For testing -->
```

Comment:
This comment is a bit weird since `ZstdFactory` does reference the `org.apache.kafka.common.record.BufferSupplier`.

Reply:
That's a good point. It probably means we need to move BufferSupplier to another package to avoid circular dependencies.

Reply:
Addressed this.